### ISSUE TITLE
Support capped CRF

### ISSUE THREAD

**Issue by JSONPARS** [google/shaka-streamer#51]
Hi,

capped CRF or constrained quality is a good way to set a quality target for encoding with an upper limit for the bitrate.
FFMPEG itself and all supported codecs natively support some kind of constrained quality.
I think it would be great if this was also available in a config file of Shaka streamers.

**Comment by joeyparrish** [comment]
Sounds reasonable.  So you would want a setting that translates to "-crf" in ffmpeg commands?

**Comment by JSONPARS** [comment]
Exactly! And also the maximum bitrate and the size of the buffer. Then you would have all parameters needed for capped CRF.

**Comment by joeyparrish** [comment]
We already have a way to control bitrate.  Does the buffer size need to be independent of the bitrate?  Does CRF need to vary per resolution?

**Comment by JSONPARS** [comment]
Yes and yes. The size of the buffer controls how large the bitrate jumps may be and thus how efficient the quality oriented encoding is.
One CRF per resolution would be important to encode smaller resolutions with higher quality.
